components/Auction.tsx

```diff
@@ -15,6 +16,8 @@ export default function Auction({
 }) {
   const [data, setData] = useState<AuctionData | null | undefined>(auctionData);
   const [pendingBid, setPendingBid] = useState<PendingBid | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [slideOverContent, setSlideOverContent] = useState<JSX.Element | null>(null);
```

Comment:
Storing JSX inside React state is usually not a good idea. This is because it breaks top-down flow of updates. When you have JSX in state, it won't behave how you'd expect React components to behave when parent props change.

Instead, think of what's the minimal amount of information you can store.

In this case, you can store an object that looks like this:


```
type SlideOverContent =  { type: 'noun', nounId: number} | { type: 'bidder', address: string };
```

(maybe instead of number/string you can store the full Noun/Bidder objects, that's fine too, as long as it's not JSX)

then you don't need `isOpen` and `slideOverContent`, you just need

```
const [slideOver, setSlideOver] = useState<SlideOverContent | null>(null);
```

Now `AuctionHeader` doesn't need to know about slide over, `onOpen`, etc. All it has is a callback prop 

```
onNounClick={() => setSlideOver({type: 'noun', noun: …})}
```

same for `BidsTable`, it shouldn't know about slideover much, it should just have

```
onBidderClick={() => setSlideOver({type: 'bidder', bidder: …})}
```